pilot/cmd/pilot-agent/main.go

```diff
@@ -301,7 +303,9 @@ var (
 					opts := make(map[string]string)
 					opts["PodName"] = podNameVar.Get()
 					opts["PodNamespace"] = podNamespaceVar.Get()
-
+					if proxyIPv6 {
+						opts["IPv6Proxy"] = "true"
```

Comment:
pass this as metadata to pilot. ISTIO_META_IP_VERSION=v6 or something..

Reply:
Hmm, I need to pass a variable when template gets rendered, there is no access to metadata in templating engine. Generating correct static entries by template is important for ipv6 to work properly. Could you please elaborate a bit on what you are proposing?

Reply:
take a look at the code that parses all environment variables starting with ISTIO_META . It takes those variables and sticks them in as node metadata

Reply:
I was talking about template rendering, input parameters are passed in `opts` map. I am failing to see how ISTIO_META is relevant to static template rendering part.. Appreciate a discussion on a slack so I could better understand.

Reply:
ISTIO_META is for passing stuff from environment ( injector, mesh expansion, etc). 
If you use something in the template - that's correct, and is not visible in Pilot. 

AFAIK pilot gets the full list of IPs of the sidecar - so no need to add another meta for pilot.